modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/UserAccountResourceTest.java

```diff
@@ -1837,6 +1840,67 @@ private void _testGetUserAccountsPageWithCustomFields() throws Exception {
 			userAccount);
 	}
 
+	private void _testGetUserAccountWithMoreExternalReferenceCodes()
+		throws Exception {
+
+		User user = UserTestUtil.addUser();
+
+		_userLocalService.addRoleUser(_role.getRoleId(), user.getUserId());
+
+		_userLocalService.addOrganizationUser(
+			_organization.getOrganizationId(), user.getUserId());
+
+		_userLocalService.addUserGroupUser(
+			_userGroup.getUserGroupId(), user.getUserId());
+
+		_userLocalService.addGroupUser(
+			testGroup.getGroupId(), user.getUserId());
+
+		UserAccount userAccount = userAccountResource.getUserAccount(
+			user.getUserId());
+
+		OrganizationBrief[] organizationBriefs =
+			userAccount.getOrganizationBriefs();
+
+		Assert.assertTrue(organizationBriefs.length > 0);
+
+		OrganizationBrief organizationBrief = organizationBriefs[0];
+
+		Assert.assertEquals(
+			_organization.getExternalReferenceCode(),
+			organizationBrief.getExternalReferenceCode());
+
+		RoleBrief[] roleBriefs = userAccount.getRoleBriefs();
+
+		Assert.assertTrue(roleBriefs.length > 1);
+
+		RoleBrief roleBrief = roleBriefs[1];
+
+		Assert.assertEquals(
+			_role.getExternalReferenceCode(),
+			roleBrief.getExternalReferenceCode());
+
+		SiteBrief[] siteBriefs = userAccount.getSiteBriefs();
+
+		Assert.assertTrue(siteBriefs.length > 0);
```

Comment:
this is wrong: because here we need to take the siteBriefs[1] we need to be sure the length is > 1